src/marionette/transport.cr

```diff
@@ -71,11 +72,24 @@ module Marionette
       Message.new(type.as_i, id.as_i, command.as_s?, params)
     end
 
+    private def get_string
+      3.times do |i|
+        s = @socket.gets(':')
+        Log.trace { "(#{@id}) Marionette receive message id: #{@last_id}, timeout #{@timeout} after #{i} tries" }
```

Comment:
we don't have timeout here, right?
```suggestion
        Log.trace { "(#{@id}) Marionette received message id: #{@last_id} after #{i} tries" }
```

Reply:
why?@socket.read_timeout = @timeout ln 22

Reply:
it is a bit confusing. If we get a message from the socket we still have the log that message is received, but there was some timeout after 1st attempt.
We have TimeoutError exception handling to log messages about that.